tvloader/parser2v1/parse_package_test.go

```diff
@@ -1069,3 +1070,38 @@ func TestFailsPackageExternalRefWithInvalidFormat(t *testing.T) {
 	}
 }
 
+func TestParser2_1PackageWithoutSpdxIdentifierThrowsError(t *testing.T) {
+	// More than one package , the previous package doesn't contain the SPDXID
+	pkgOldName := "p1"
+	parser := tvParser2_1{
+		doc: &spdx.Document2_1{Packages: map[spdx.ElementID]*spdx.Package2_1{}},
+		st:  psPackage2_1,
+		pkg: &spdx.Package2_1{PackageName: pkgOldName},
+	}
+	pkgOld := parser.pkg
+	parser.doc.Packages["p1"] = pkgOld
+	// the Document's Packages should have this one only
+	if parser.doc.Packages["p1"] != pkgOld {
+		t.Errorf("expected package %v, got %v", pkgOld, parser.doc.Packages["p1"])
+	}
+	if len(parser.doc.Packages) != 1 {
+		t.Errorf("expected 1 package, got %d", len(parser.doc.Packages))
+	}
+
+	// Case 2: Checks the Last package
+	pkgName := "p2"
+	err := parser.parsePair2_1("PackageName", pkgName)
+	if err == nil {
+		t.Errorf("packages withoutSpdx Identifiers getting accepted")
+	}
+	tvPairs := []reader.TagValuePair{
```

Comment:
This appears to be where Case 2 starts. As with the similar code in parse_file_test.go, I think this belongs in parser_test.go instead. It looks like this is actually testing code that occurs at the end of parsing via parser.go. Can you move this out into parser_test.go?